bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/IdentityTokenSupplier.java

```diff
@@ -23,73 +22,41 @@ public class IdentityTokenSupplier implements Serializable {
       "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=%s&format=%s&licenses=%s";
   private static final String METADATA_FLAVOR_HEADER_KEY = "Metadata-Flavor";
   private static final String METADATA_FLAVOR_HEADER_VALUE = "Google";
-  private static final long ttlInMillis = 50 * 60 * 1000;
   private static final RequestConfig config =
       RequestConfig.custom()
           .setConnectTimeout(100)
           .setConnectionRequestTimeout(100)
           .setSocketTimeout(100)
           .build();
-  private final String audience;
   private static final String format = "full";
   private static final boolean license = true;
-  private long lastFetchTime;
-  private String cachedToken;
 
-  public IdentityTokenSupplier(String audience) {
-    this.audience = audience;
-    cachedToken = null;
-    lastFetchTime = 0;
-  }
-
-  @Override
-  public int hashCode() {
-    return Objects.hashCode(cachedToken, lastFetchTime);
-  }
+  public static Optional<String> createIdentityToken(String audience) {
+    CloseableHttpClient httpClient = HttpClients.custom().setDefaultRequestConfig(config).build();
```

Comment:
Have we looked into https://cloud.google.com/docs/authentication/get-id-token#java?